playbooks/webhooks.yml

```diff
@@ -49,37 +63,104 @@
   hosts: job_nodes
   user: root
   vars:
-    github_cidr: "192.30.252.0/22"
-  pre_tasks:
-    - name: Install apt packages
-      apt:
-        pkg: "apache2-utils"
-        state: installed
-        update_cache: yes
+    GITHUB_CIDR: "192.30.252.0/22"
+    GITHUB_HOOK_PORT: 80
 
-    # The nginx instance for basic auth listens on 80
-    # the ssh tunnels listens on 8888 but is not open to the public.
-    - name: Allow webhooks
-      ufw:
-        rule: allow
-        port: 80
-        src: "{{github_cidr}}"
+    RPC_GATING_REPO: "{{ lookup('env', 'RPC_GATING_REPO') }}"
+    RPC_GATING_BRANCH: "{{ lookup('env', 'RPC_GATING_BRANCH') }}"
+    RPC_GATING_DIR: /opt/rpc-gating
 
+    WHT_DIR: "{{RPC_GATING_DIR}}/webhooktranslator"
+    WHT_VENV_DIR: /opt/venv_webhooktranslator
+    WHT_SOCK: /run/webhooktranslator.sock
+    WHT_PORT: 5000
+
+    RACKSPACE_WEBHOOK_TOKEN: "{{ lookup('env', 'RACKSPACE_WEBHOOK_TOKEN') }}"
+    JIRA_USER: "{{ lookup('env', 'JIRA_USER') }}"
+    JIRA_PASSWORD: "{{ lookup('env', 'JIRA_PASS') }}"
+    JIRA_PROJECT: "{{ lookup('env', 'JIRA_PROJECT') }}"
+
+  pre_tasks:
+    # used by webhooks.groovy to create the ssh tunnel script
     - name: Write ip file
       delegate_to: localhost
       copy:
         dest: "{{WORKSPACE}}/instance_address"
         content: "{{ansible_default_ipv4.address}}"
 
+    # Setup webhook translator
+
+    - name: Install virtualenv tool
+      pip:
+        name: "virtualenv"
+
+    - name: Clone rpc-gating
+      git:
+        repo: "{{ RPC_GATING_REPO }}"
+        version: "{{ RPC_GATING_BRANCH }}"
+        dest: "{{ RPC_GATING_DIR }}"
+
+    - name: Install webhooktranslator
+      pip:
+        virtualenv: "{{ WHT_VENV_DIR }}"
+        name: "{{ WHT_DIR }}"
```

Comment:
Setuptools sets the execute bit when it creates the console scripts as install time.